### PR TITLE
Authorization rules

### PR DESCRIPTION
**Pull request by chepkok3**
In this PR, we:

- Installed CanCanCan in the project.
- Added a role column to the user's table. 
- [ ] Ensured that:
- A user can delete a post if it is theirs or if they have an admin role using CanCanCan for authorization.
- A user can delete a comment if it is theirs or if they have an admin role using CanCanCan for authorization.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-  end
-  resources :posts, only: %i[index show new create] do
-    resources :comments, only: %i[new create]
-    resources :likes, only: [:create]
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  # root "articles#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create, :destroy] do
```
Comment (DammyShittu) [inline config/routes.rb:8]: - [ ] When I visit the post new path I get this error page. Kindly fix that. See comment below

  image

File: app/controllers/posts.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class PostsController < ApplicationController
```
Comment (DammyShittu) [inline app/controllers/posts.rb]: - [ ] This file name should be `posts_controller.rb`, not `posts.rb`. Kindly rename it.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @chepkok3 and @hienphan0111,

## Highlights
✔️ Cancancan for authorization

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts.rb (hunk 2 above)
Comment (chepkok3) [reply app/controllers/posts.rb]: @DammyShittu Sure it should posts_controller.rb. However, I do not understand how the error you got came about, can you please explain it further?

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**